blaze/expr/collections.py

```diff
@@ -100,10 +100,32 @@ def sort(child, key=None, ascending=True):
     ascending : bool, optional
         Determines order of the sort
     """
+    if ascending not in (True, False):
+        msg = "ascending must be True or False, given {}"
+        raise ValueError(msg.format(ascending))
     if not isrecord(child.dshape.measure):
-        key = None
+        if key is None or isinstance(key, _strtypes):
+            # Handle this case separately.
+            return Sort(child, None, ascending)
```

Comment:
if it is not a record should we just raise if we sort with a key, like, `TypeError('Cannot sort a non-record by a column name')`?